Added TTL 74670 4-by-4 register file with three-state outputs

BunchOfBits opened the pull request:
Added TTL 74670 4-by-4 register file with three-state outputs

maehne commented on src/main/resources/resources/logisim/strings/std/std_de.properties:
```diff
@@ -813,6 +813,7 @@ TTL7454 = 7454: Vierfach UND-OR-INVERT-Gatter
 TTL74541 = 74541: Achtfach Puffer / Leitungstreiber
 TTL7458 = 7458: Zweifach AND-OR Gatter
 TTL7464 = 7464: 4-2-3-2-2 AND-OR-INVERT-Gatter
+# ==> TTL74670 =
```
4-mal-4-Bit-Registerspeicher mit Tri-State-Ausgängen

BunchOfBits replied:
Done. Thanks once again for your translation

maehne commented on src/main/resources/resources/logisim/strings/std/std_fr.properties:
```diff
@@ -813,6 +813,7 @@ TTL7454 = 7454 : Quatre portes ET-OU-OU-INVERTES larges
 TTL74541 = 74541 : Huit tampons / circuits d’excitation
 TTL7458 = 7458 : porte double AND-OR
 TTL7464 = 7464 : 4-2-3-3-2 Porte AND-OR-INVERT
+# ==> TTL74670 =
```
banc de registre 4 fois 4 bits avec sorties tri-état

BunchOfBits replied:
Done. Thanks for your french translation.

maehne reviewed (dismissed):
Looks good to me! Your modifications to ClockState.java would deserve some more comments.

R3dst0ne commented:
I just noticed that you feature the line `Original code by Marcin Orlowski (http://MarcinOrlowski.com), 2021.` in the header of each of your TTL PRs. 
You probably should change that.

BunchOfBits commented:
> I just noticed that you feature the line `Original code by Marcin Orlowski (http://MarcinOrlowski.com), 2021.` in the header of each of your TTL PRs. You probably should change that.

Is there a recommended header ? Honestly, I've just copied it from one file to another, like you probably already suspected

R3dst0ne commented:
To my knowledge, there is no recommended header. (Please correct me if I'm wrong.)

In my experience
> /*
> \* Logisim-evolution - digital logic design tool and simulator
> \* Copyright by the Logisim-evolution developers
> \*
> \* https://github.com/logisim-evolution/
> \*
> \* This is free software released under GNU GPLv3 license
> */

is the most used one.

There is an issue (#1264) and a PR (#1276) regarding this topic.

It is probably best to ask @BFH-ktt1 regarding this type of questions.

maehne commented on src/main/java/com/cburch/logisim/std/ttl/Ttl74670.java:
```diff
@@ -0,0 +1,208 @@
+/*
+ * This file is part of logisim-evolution.
+ *
+ * Logisim-evolution is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Logisim-evolution is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with logisim-evolution. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Original code by Marcin Orlowski (http://MarcinOrlowski.com), 2021.
```
Please adjust according to @R3dst0ne's recommendation.

BunchOfBits replied:
Done. I've used the copyright message which @R3dst0ne  suggested

maehne reviewed (changes requested):
Good except for the file header.

BunchOfBits commented:
Two things. 1) I will replace the copyright messages on the files that I've added in the past. 2) I will also update the changes.md file about all chips that I've added to the TTL library. I've noticed that some have slipped through the cracks and I will fix that in a separate PR.

BunchOfBits commented:
Is there anything left to be done for this PR ? (apart from the things I mentioned that I will do in a separate PR). If not, can we merge this PR ?

maehne commented:
@BunchOfBits: All fine from my side now! Thanks for your contributions!
